### ISSUE TITLE
As a stubserver consumer I want to be able to specify the behavior of the stub latency as a random distribution

### ISSUE THREAD

**Issue by TimSoethout** [scala-academy/castalia#5]
Acceptance criteria:

background:
given that the stubserver is started

given that I configured a stub to use a normal distributed latency with specified p95 and p99 values
when I get a lot of response from that stub
then the latency of those responses should coincide that normal distribution

**Comment by jordi133** [comment]
Perhaps saying that the performance characteristics are configurable is better than saying that it should be "the same"

**Comment by jordi133** [comment]
Example json:

{
  "endpoint": "jsonconfiguredstub",
  "responses": [
    { 
      "id": "1",
      "latency": "50 ms",
      "httpStatusCode": 200,
      "response": {
        "id": "een",
        "someValue": "123123"
      }
    },
    { 
      "id": "2",
      "latency": "0 ms",
      "httpStatusCode": 200,
      "response": {
        "id": "twee",
        "someValue": "123123",
        "someAdditionalValue": "345345"
      }
    },
    { 
      "id": "0",
      "latency": "2 s",
      "httpStatusCode": 404
    }
  ]
}

**Comment by effibennekers** [comment]
Please describe normal latency in terms of p95 and p99
Please describe uniform latency in terms of lowerBound and upperBound

**Comment by jordi133** [comment]
Example json:

{
  "endpoint": "jsonconfiguredstub",
  "responses": [
    { 
      "id": "1",
      "delay": {
        "distribution": "normal",
        "p95": "500 ms"
        "p99": "750 ms"
      },
      "httpStatusCode": 200,
      "response": {
        "id": "een",
        "someValue": "123123"
      }
    },
    { 
      "id": "2",
      "httpStatusCode": 200,
      "response": {
        "id": "twee",
        "someValue": "123123",
        "someAdditionalValue": "345345"
      }
    },
    { 
      "id": "0",
      "delay": {
        "distribution": "normal",
        "p95": "100 ms"
        "p99": "2000 ms"
      },
      "httpStatusCode": 404
    }
  ]
}

**Comment by jeanmarc** [comment]
What do you consider a 'normal' distribution for delays/latency? The only definition of normal distribution that I know is a bell-shaped curve from -Inf to +Inf that is centered on a mean value, and has a standard deviation. The p95 of such a distribution is calculated as mean + 1.65 \* standard deviation. Similar calculation for p99.
However, in practice latency does not have a normal distribution, as there can never be negative values.
Maybe better suited is a Gamma Distribution, which defines a probability on [0, inf> and depending on the parameters has a high probability for small values, and a low probability for larger values.

How about using as input: 
- min, max == always at least min, always at most max (+ some millis)
- distribution:
  *\* uniform (all latencies between min and max are equally likely to occur)
  *\* normal (uses (min+max)/2 as mean, min and max are on 3 \* sd) 
  *\* gamma (needs a 'shape' and 'scale' parameter)

The gamma version will then generate latencies between min and max, according to the shape/scale specified. Gamma function is stretched from min to max, with max == p99 of the distribution.

**Comment by DieBauer** [comment]
I agree, normal distribution makes little sense in delays when looked at mathematically. We could however filter all negatives(non-physical delays), making it not a normal distribution anymore. In the feature/delay-trait i already added a library that can generate values with a gamma distribution. Worth a try.
